src/Symfony/Component/Workflow/Dumper/GraphvizDumper.php

```diff
@@ -0,0 +1,262 @@
+ */
+
+namespace Symfony\Component\Workflow\Dumper;
+
+use Symfony\Component\Workflow\Definition;
+use Symfony\Component\Workflow\Marking;
+
+/**
+ * GraphvizDumper dumps a workflow as a graphviz file.
+ *
+ * You can convert the generated dot file with the dot utility (http://www.graphviz.org/):
+ *
+ *   dot -Tpng workflow.dot > workflow.png
+ *
+ * @author Fabien Potencier <fabien@symfony.com>
+ * @author Grégoire Pineau <lyrixx@lyrixx.info>
+ */
+class GraphvizDumper implements DumperInterface
+{
+    private static $defaultOptions = array(
+        'graph' => array('ratio' => 'compress', 'rankdir' => 'LR'),
+        'node' => array('fontsize' => 9, 'fontname' => 'Arial', 'color' => '#333333', 'fillcolor' => 'lightblue', 'fixedsize' => true, 'width' => 1),
+        'edge' => array('fontsize' => 9, 'fontname' => 'Arial', 'color' => '#333333', 'arrowhead' => 'normal', 'arrowsize' => 0.5),
+    );
+
+    /**
+     * Dumps the workflow as a graphviz graph.
+     *
+     * Available options:
+     *
+     *  * graph: The default options for the whole graph
+     *  * place: The default options for places
+     *  * edge: The default options for edges
+     *
+     * @param array      $options    An array of options
+     * @param Definition $definition A Definition instance
+     *
+     * @return string The dot representation of the workflow
+     */
+    public function dump(Definition $definition, Marking $marking = null, array $options = array())
+    {
+        $places = $this->findPlaces($definition, $marking);
+        $transitions = $this->findTransitions($definition);
+        $edges = $this->findEdges($definition);
+
+        $options = array_replace_recursive(self::$defaultOptions, $options);
+
+        return $this->startDot($options)
+            .$this->addPlaces($places)
+            .$this->addTransitions($transitions)
+            .$this->addEdges($edges)
+            .$this->endDot();
+    }
+
+    /**
+     * Finds all places.
+     *
+     * @return array An array of all places
+     */
+    private function findPlaces(Definition $definition, Marking $marking = null)
+    {
+        $places = array();
+
+        foreach ($definition->getPlaces() as $place) {
+            $attributes = [];
+            if ($place === $definition->getInitialPlace()) {
+                $attributes['style'] = 'filled';
+            }
+            if ($marking && $marking->has($place)) {
+                $attributes['color'] = '#FF0000';
+                $attributes['shape'] = 'doublecircle';
+            }
+            $places[$place] = array(
+                'attributes' => $attributes,
+            );
+        }
+
+        return $places;
+    }
+
+    /**
+     * Finds all transitions.
+     *
+     * @return array An array of all transitions
+     */
+    private function findTransitions(Definition $definition)
+    {
+        $transitions = array();
+
+        foreach ($definition->getTransitions() as $name => $_) {
```

Comment:
I know it's not used, but IMO this should follow CS & that variable (`$_`) should have normal name ;)